crates/credentials/src/vc.rs

```diff
@@ -127,25 +195,163 @@ impl VerifiableCredential {
         Ok(jwt)
     }
 
-    pub async fn verify(jwt: &str) -> Result<Self, CredentialError> {
+    pub async fn verify(jwt: &str) -> Result<Arc<VerifiableCredential>, CredentialError> {
         let jwt_decoded = Jwt::verify::<VcJwtClaims>(jwt).await?;
+        let mut vc = jwt_decoded.claims.vc;
+        let registered_claims = jwt_decoded.claims.registered_claims;
 
-        // TODO Implement semantic VC verification rules https://github.com/TBD54566975/web5-rs/issues/151
+        // registered claims checks
+        if registered_claims.issuer.is_none() {
+            return Err(CredentialError::MissingIss);
+        }
+        if registered_claims.subject.is_none() {
+            return Err(CredentialError::MissingSub);
+        }
+        if registered_claims.not_before.is_none() {
+            return Err(CredentialError::MissingNbf);
+        }
+        if registered_claims.jti.is_none() {
+            return Err(CredentialError::MissingJti);
+        }
 
-        Ok(jwt_decoded.claims.vc)
+        // TODO: Change this to None after this issue is resolved - https://github.com/TBD54566975/web5-rs/issues/202
+        // check claims match expected values
+        if vc.id == String::default() {
+            vc.id = registered_claims.jti.clone().unwrap();
```

Comment:
so technically `unwrap` should be safe here since we're checking `if registered_claims.jti.is_none()` above, but I think it's ill advised to ever use `unwrap` or `expect` (or a number of other methods) in non-test code because there is the potential for a panic (any invocation which could panic should be avoided). instead, you could implement this one of a few ways, my recommended approach would be to remove the `if registered_claims.jti.is_none()` block above and handle that inline here like so:

```suggestion
            vc.id = registered_claims.jti.ok_or(CredentialError::MissingJti)?;
```

the same would hold true for all of the surrounding code which follows the same pattern

there's also a way to implement it using a `match` block

Reply:
much better